### PR TITLE
Add SMART self-test results to zpool status -c

### PR DESCRIPTION
**Pull request by tonyhutter**
### Description
Add in SMART self-test results to `zpool status|iostat -c`.  This works for both SAS and SATA drives.

```
$ zpool status -c smart_test
  pool: mypool
 state: ONLINE
  scan: none requested
config:

	NAME           STATE     READ WRITE CKSUM         test_type             test_status  test_progress  test_ended
	mypool         ONLINE       0     0     0
	  /tmp/file1   ONLINE       0     0     0  background_short                 aborted              -           -
	  /tmp/file2   ONLINE       0     0     0  extended_offline               completed           100%       9d10h
	  /tmp/file3   ONLINE       0     0     0     short_offline               completed           100%           -
	  /tmp/file4   ONLINE       0     0     0  extended_offline               completed           100%           -
	  /tmp/file5   ONLINE       0     0     0   background_long               completed              -     1347d4h
	  /tmp/file6   ONLINE       0     0     0  background_short                 aborted              -           -
	  /tmp/file7   ONLINE       0     0     0   background_long                 running            90%           -
	  /tmp/file8   ONLINE       0     0     0  extended_offline  completed:read_failure            90%          4h
	  /tmp/file9   ONLINE       0     0     0   background_long                 running            90%           -
	  /tmp/file10  ONLINE       0     0     0   background_long                 running            90%           -
```
```
smart_test:	Show SMART self-test results summary.
test_type:	Show SMART self-test type (short, long... ).
test_status:	Show SMART self-test status.
test_progress:	Show SMART self-test percentage done.
test_ended:	Show when the last SMART self-test ended (if supported).
```
### Motivation and Context
We plan to run SMART self-tests on all drives that  are showing SMART or ZFS errors, but are not yet faulted.  This will allow us to easily see the test results.

### How Has This Been Tested?
Tested using 12 different samples of smartctl output from SAS and SATA drives at various stages of self-test.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux code style requirements.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain `Signed-off-by`.
- [ ] Change has been approved by a ZFS on Linux member.

### CODE REVIEW

**Comment by bunder2015** [comment]
Quick question, for SATA does this cover attribute 194 "Temperature_Celsius" as well as 190 "Airflow_Temperature_Cel"?  The latter is what is returned on Samsung SSD's.  I also have a Samsung NVME which presents 3 temperatures ("Temperature", "Temperature Sensor 1" and 2, no attribute numbers shown however)

**Comment by tonyhutter** [comment]
Thanks.  I'm planning to rebase after https://github.com/zfsonlinux/zfs/issues/7183 is checked in since it will conflict.

File: cmd/zpool/zpool.d/smart [hunk 1]
```diff
@@ -45,7 +50,7 @@ if [ -b "$VDEV_UPATH" ] && [ -x "$smartctl_path" ]; then
 	# SATA:
 	#	ATA Version is:   8
 	#
-	type=$(echo "$raw_out" | grep -m 1 -Eo '^ATA|SAS$')
+	type=$(echo "$raw_out" | grep -m 1 -Eo '^ATA|SAS')
```
Comment (bunder2015) [inline cmd/zpool/zpool.d/smart]: I've been thinking about this.  If we set type in the awk section, do we need to here?  We might want to revise the -z type section below as well.

Comment (tonyhutter) [reply cmd/zpool/zpool.d/smart]: Yea, we could grep `type`  from `$out` so that we're only setting it in the awk section.

**Comment by tonyhutter** [comment]
@h1z1 yea, we should probably add a `Recommends: smartmontools` line to the spec for the distros that support it, and print an error if you can't run smartctl for any reason.  Error reporting for `-c` is basically non-existent right now...  We do have a note about adding smartctl to the sudoers file, but looking at it now, I don't think it will work for NVMe devices:
```
$ cat etc/sudoers.d/zfs 
##
## Allow any user to run `zpool iostat/status -c smart` in order
## to read basic SMART health statistics for a pool.
##
## CAUTION: Any syntax error introduced here will break sudo.
##

# ALL ALL = (root) NOPASSWD: /usr/sbin/smartctl -a /dev/[hsv]d[a-z0-9]*
```
So lots of work still needed...

**Comment by h1z1** [comment]
@tonyhutter Indeed there is the formatting and device regex not matching but also allowing users to do stupid things like calling smartctl in a loop.  It will pound the drive(s) in question.  Discloses serial numbers, etc too.

**Pull request merged**